.github/workflows/verify-templates.yml

```diff
@@ -6,50 +6,30 @@ on:
     - cron: '0 14 * * *'
 jobs:
 
-  angular-build:
-    name: Build Angular project
+  build:
+    name: Test generated project
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        frontend: [angular, vue, react]
     steps:
       - name: Set up Go
         uses: actions/setup-go@v1
         with:
           go-version: 1.14
-        id: go
       - name: Checkout project
         uses: actions/checkout@v2
-      - name: Generate Angular project
-        run: go run main.go init --frontend angular app
-      - name: Build the Docker image
-        run: cd app && docker build .
-
-  react-build:
-    name: Build React project
-    runs-on: ubuntu-latest
-    steps:
-      - name: Set up Go
-        uses: actions/setup-go@v1
-        with:
-          go-version: 1.14
-        id: go
-      - name: Checkout project
-        uses: actions/checkout@v2
-      - name: Generate React project
-        run: go run main.go init --frontend react app
-      - name: Build the Docker image
-        run: cd app && docker build .
-
-  vue-build:
-    name: Build Vue project
-    runs-on: ubuntu-latest
-    steps:
-      - name: Set up Go
-        uses: actions/setup-go@v1
-        with:
-          go-version: 1.14
-        id: go
-      - name: Checkout project
-        uses: actions/checkout@v2
-      - name: Generate Vue project
-        run: go run main.go init --frontend vue app
-      - name: Build the Docker image
-        run: cd app && docker build .
+      - name: Generate application
+        run: go run main.go init --frontend ${{ matrix.frontend }} app
+      - name: Start the application
+        run: cd app && docker-compose up -d
+      - name: Check availability of backend
+        run: timeout 10 bash -c 'until curl -s localhost:8080/api/technologies | grep Go > /dev/null; do sleep 1; done'
+      - name: Check availability of frontend
+        run: |
+          npm i -g domcurl
```

Comment:
This was the easiest solution to test the rendered frontend page, but relying on an npm package for that doesn't seem ideal to me. What do you think? @Shpota

Reply:
It is not ideal, but I don't know a better way to query dynamic content. 

At the same time, it is a part of CI code. In my opinion, it is fine if it is not perfect while it does its job.

We can improve it it once we find a better way.

Reply:
Why don't you use this opportunity to do very basic E2E tests? Rather than relying on an NPM package that hardly anyone knows about, you can use an industry-standard E2E testing tool like Selenium to ensure the Goxygen template is rendering correctly. Since its render involves querying the Go backend, you can verify just from the Selenium tests that both the app is rendering correctly, and the backend is responding correctly.

Using Selenium, however, does have overhead. Having E2E tests never hurt anyone, though.

Reply:
Good point we could definetly elaborate on this. Do you have any suggestion about which tooling to use for the e2e tests?

Reply:
I mentioned Selenium in my last comment, however it's probably not the best bet as we'll be testing frontend frameworks, and not plain HTML pages. I've worked with Selenium wrappers testing React, and it's a huge pain. I won't elaborate why here, but if you're curious I can elaborate later.

[Cypress](https://www.cypress.io/) is an E2E testing framework that's built for the modern web, and it is also widely used/supported/documented.